flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/DataFormatConverters.java

```diff
@@ -164,9 +164,10 @@ public static DataFormatConverter getConverterForDataType(DataType originDataTyp
 				Tuple2<Integer, Integer> ps = getPrecision(logicalType);
 				if (clazz == BigDecimal.class) {
 					return new BigDecimalConverter(ps.f0, ps.f1);
-				} else {
+				} else if (clazz == Decimal.class) {
 					return new DecimalConverter(ps.f0, ps.f1);
 				}
+				throw new RuntimeException("Not support class for DECIMAL: " + clazz);
```

Comment:
Put it in the `else` branch? Could you help to update other types which have the same problem, e.g. `TIMESTAMP_WITHOUT_TIME_ZONE`, `TIMESTAMP_WITH_LOCAL_TIME_ZONE`, `TIMESTAMP_WITH_LOCAL_TIME_ZONE`...